### DISCUSSION TITLE
How to use Fastapi without Sqlachemy

### DISCUSSION THREAD

**Discussion by scheung38** [fastapi/fastapi#8936] [Questions]
Any examples of usage without SQLALCHEMY please? I just want to make existing def that uses psycopg2 to be wrapped with FastAPI without SQLALCHEMY as all the queries are already present and I do not want to use ORM

**Comment by Mause** [comment]
```python
import psycopg2
from fastapi import Depends, FastAPI


def connection():
    return psycopg2.connect("dbname=test user=postgres")


@app.get('/')
def index(conn = Depends(connection)):
    cursor = conn.cursor()
    cursor.execute('select * from test')
    return cursor.fetchall()
```

Is that what you were looking for?

**Comment by scheung38** [comment]
Thanks @Mause when to use async await and when not to thanks, why sometimes we see async but no corresponding await associated with it?

**Comment by Mause** [comment]
@scheung38 see the documentation: https://fastapi.tiangolo.com/async
